components/my-account/User/UserForm/UserForm.tsx

```diff
@@ -0,0 +1,368 @@
+
+interface UserFormProps {
+  isEditMode: boolean
+  b2BUser?: B2BUser | any
+  closeUserForm: () => void
+}
+
+export const useFormSchema = () => {
+  const { t } = useTranslation('common')
+  return yup.object({
+    emailAddress: yup
+      .string()
+      .required(t('no-email-error'))
+      .matches(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i, t('invalid-email-error')),
+    firstName: yup.string().required(t('firstname-error')),
+    lastName: yup.string().required(t('lastname-error')),
+  })
+}
+
+const UserForm = (props: UserFormProps) => {
+  const { isEditMode, b2BUser, closeUserForm } = props
+
+  const { publicRuntimeConfig } = getConfig()
+
+  const { t } = useTranslation('common')
+  const theme = useTheme()
+  const mdScreen = useMediaQuery(theme.breakpoints.up('md'))
+  const userSchema = useFormSchema()
+  const { user } = useAuthContext()
+  const userRoles = publicRuntimeConfig.b2bUserRoles
+  const userFormRadioOptions = publicRuntimeConfig.userFormRadioOptions
+  const { createCustomerB2bUser } = useCreateCustomerB2bUserMutation()
+  const { addRoleToCustomerB2bAccount } = useAddRoleToCustomerB2bAccountMutation()
+  const { updateCustomerB2bUser } = useUpdateCustomerB2bUserMutation()
+  const { deleteB2bAccountUserRole } = useDeleteB2bAccountRoleMutation()
+
+  const [isLoading, setLoading] = useState(false)
+  const {
+    getValues,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+    control,
+    reset,
+  } = useForm({
+    defaultValues: { role: '', emailAddress: '', firstName: '', lastName: '', isActive: true },
+    resolver: yupResolver(userSchema),
+  })
+  const onSubmit = async () => {
+    if (isLoading) return
+    setLoading(true)
+    const formValues = getValues()
+    if (isEditMode) {
+      const variables = buildUpdateCustomerB2bUserParams({ user, b2BUser, values: formValues })
+      const updateUserResponse = await updateCustomerB2bUser.mutateAsync({
+        ...variables,
+      })
+      const previousRole = b2BUser?.roles[0]?.roleName
+      if (updateUserResponse && b2BUser?.roles.length && formValues.role !== previousRole) {
+        await deleteB2bAccountUserRole.mutateAsync(
+          buildB2bUserRoleParams({
+            user,
+            b2BUser,
+            values: { role: previousRole },
+            roles: userRoles,
+          })
+        )
+      }
+      addRoleToB2bUser(updateUserResponse, formValues)
+    } else {
+      const variables = buildCreateCustomerB2bUserParams({ user, values: formValues })
+      const createUserResponse = await createCustomerB2bUser.mutateAsync({
+        ...variables,
+      })
+      if (createUserResponse?.userId) {
+        addRoleToB2bUser(createUserResponse, formValues)
+      }
+    }
+    setLoading(false)
+    cancelAction()
+    // }
+  }
+
+  useEffect(() => {
+    if (!b2BUser) return
+    const { firstName, lastName, emailAddress, isActive, roles } = b2BUser
+    setValue('emailAddress', emailAddress)
```

Comment:
I may be wrong but can't we use react-hook-form default values for this operation ?